src/client/components/Checkbox/Checkbox.js

```diff
@@ -7,36 +7,44 @@ import './Checkbox.css'
 export type Props = {
   checked: boolean,
   value: any,
-  onCheck?: (e: SyntheticInputEvent, value: boolean) => any,
+  onChange?: (e: SyntheticInputEvent) => void,
   name: string,
   disabled?: boolean,
 }
 
 class Checkbox extends Component {
-  /* eslint-disable react/sort-comp */
   constructor(props: Props) {
-    super(props);
+    super(props)
+    this.state = {
+      checked: this.props.checked,
+    };
     (this:any).handleClick = this.handleClick.bind(this)
   }
 
+  state: {
+    checked: boolean,
+  }
+
   props: Props
 
   handleClick(e: SyntheticInputEvent) {
-    if (this.props.onCheck) {
-      this.props.onCheck(e, e.target.checked)
+    if (this.props.onChange) {
+      this.setState({
```

Comment:
1) you should not check if onChange handler exists to change the state, do it before check, otherwise checkbox will never work without onChange handler

2) but if handler exists you should trigger it otherwise how would we subscribe on change event if it's never triggered